src/main/scala/org/apache/pekko/stream/contrib/Timed.scala

```diff
@@ -87,21 +88,23 @@ object Timed extends TimedOps with TimedIntervalBetweenOps {
   final class TimedFlowContext {
     import scala.concurrent.duration._
 
-    private val _start = new AtomicLong
-    private val _stop = new AtomicLong
+    private val _started = new AtomicBoolean(false)
+    private val _start = new AtomicLong(0)
+    private val _stop = new AtomicLong(0)
 
-    def start(): Unit =
+    def start(): Unit = {
       _start.compareAndSet(0, System.nanoTime())
+      _started.set(true)
+    }
 
     def stop(): FiniteDuration = {
       _stop.compareAndSet(0, System.nanoTime())
       compareStartAndStop()
     }
 
     private def compareStartAndStop(): FiniteDuration = {
-      val stp = _stop.get
-      if (stp <= 0) Duration.Zero
```

Comment:
I'm not sure what was the point of this check, but I'm pretty sure that this is wrong.

From [the documentation](https://docs.oracle.com/javase/8/docs/api/java/lang/System.html#nanoTime--):

> The value returned represents nanoseconds since some fixed but arbitrary origin time (perhaps in the future, so values may be negative).

And indeed, I noticed this issue because I started getting negative durations for failed streams.